### ISSUE TITLE
BED files should not have headers

### ISSUE THREAD

**Issue by huddlej** [nextstrain/augur#510]
**Current Behavior**  

Wherever augur refers to BED files, it tends to do so with the expectation that these files have a header in the first line. For example, the [load_excluded_sites function in tree.py](https://github.com/nextstrain/augur/blob/d5dbc75bce43f9439bad5ac3016ded492fb43b7a/augur/tree.py#L207) and the [test data for the TB build](https://github.com/nextstrain/augur/blob/e17be1510d0c7bbcc23f67a0576ba3d9a76009c4/tests/builds/tb/data/Locus_to_exclude_Mtb.bed#L1).

[Standard BED format](https://genome.ucsc.edu/FAQ/FAQformat.html#format1) _does not_ support a header line, so augur's current behavior will lead to silent omission of the first record in each proper BED file a user provides when that record gets treated as the header.

**Expected behavior**  

If augur refers to BED format for its input files, it should not assume there will be a header line.

**Possible solution**  

 - Find all instances where augur reads in a CSV file with pandas that it expects to be a BED file and add a `header=None` argument to the pandas `read_csv` call.
 - Update all BED files in test data and live augur builds to exclude header lines

Alternately, we could retain the current format with the header and rename the expected inputs as "tab-delimited coordinates".

**Comment by tsibley** [comment]
> Alternately, we could retain the current format with the header and rename the expected inputs as "tab-delimited coordinates".

I'd vote against this. It's nice to stick to standard, widely-used formats like BED, even when the difference between our "tab-delimited coordinates" and BED would be very small.

[The number of affected files](https://github.com/search?q=user%3Anextstrain+user%3Aemmahodcroft+user%3Arneher+user%3Ablab+extension%3Abed+-repo%3Ablab%2Fzika-seq&type=Code), at least in our group, seems pretty low. So would be nice to make this actually BED.

We could also implement the change in a backwards-compat way by inspecting if the first line. The example header you linked to isn't a valid BED line, so such inspection wouldn't run the risk of false positives.

**Comment by emmahodcroft** [comment]
I'd really support backwards compatibility here. The number in our group is low because I'm the only one who works with VCF and the only 'nextstrain' VCF builds we maintain are the one TB build & tutorial. But, I have plenty of other builds that use this (just not official Nextstrain ones), and I expect other bacterial users will too.  

Happy to support both ways, but think compatibility would be very good here - this is a small thing, and personally I also think having a header makes it clearer anyway!

**Comment by huddlej** [comment]
Cool, I prefer supporting standard BED files with backward-compatibility, too. It's true that headers are better and BED is a suboptimal format.

Backward compatibility has been implemented for one instance of reading BED files by @danielsoneg in the [augur mask module](https://github.com/nextstrain/augur/blob/7258b8b4979e2ef80f8773041d930cd01abb724e/augur/mask.py#L32-L38). We've discussed moving this function (or maybe the more generic function from `tree.py`) into `utils.py` and updating augur to use this function everywhere BED files are loaded.
